LiteCore/Query/QueryParserTables.hh

```diff
@@ -68,7 +68,8 @@ namespace litecore {
         {"MATCH"_sl,   2, 2,  3,  &QueryParser::matchOp},
         {"BETWEEN"_sl, 3, 3,  3,  &QueryParser::betweenOp},
         {"EXISTS"_sl,  1, 1,  8,  &QueryParser::existsOp},
-
+        {"NULLORMISSING"_sl,     1, 1, 4, &QueryParser::nullOrMissingOp},
+        {"NOT NULLORMISSING"_sl, 1, 1, 4, &QueryParser::nullOrMissingOp},
```

Comment:
No I am suggesting this case is already handled because the NOT operator already exists and that this `NOT NULLORMISSING` is redundant.

Reply:
Conceptually, we have two postfix operators in n1ql, IS NULLORMISSING and IS NOT NULLORMISSING. They are multi-token operators,
   expression IS NULLORMISSING
   expression IS NOT NULLORMISSING
They are translated to JSON query operators, ["NULLORMISSING", expression] and ["NOT NULLORMISSING", expression]
NOT in the above is not passed as an operator; it is a lexical part of the new operator, IS NOT NULLORMISSING.
This is just easy to handle, and appears user friendly. Of course, another approach is to treat IS NOT NULLORMISSING as syntactic sugar and translate it to
expression IS NOT NULLORMISSING --> NOT (expression IS NULLORMISSING) --> ["NOT" ["NULLORMISSING" expression]]. This way, we only have one operator in the JSON query language. Then, JSON form would be less user-friendly if we think "expr IS NOT NULLORMISSING" is more preferable than "NOT (expr IS NULLORMISSING)"

Reply:
I understand that your PR makes NOT a lexical part of the new operator but my question is "why?"  To make a friendlier JSON?  Now it is inconsistent though since we don't have corresponding `NOT NULL` and `NOT MISSING` operators.  What makes this new operator special that it needs its own variant of `NOT`?

Reply:
We don't need `NOT NULLORMISSING`; I agree with Jim. The JSON can just wrap `NOT` around it.

Also, it would be simpler to make this a function, e.g. `nullormissing()`. That way QueryParser doesn't have to be updated. You just add an entry to the function table, and a corresponding C function.

Reply:
@pasin can we do away with "IS NOT NULLORMISSING" ? In N1QL, it would simply be "NOT x IS NULLORMISSING" instead of "x IS NOT NULLORMISSING".
By the way, server n1ql have all variants of IS NOT NULL, IS NOT MISSING, etc. Their "IS" is not a general operator.  "1 IS 1" is illegal. IS has to be followed by NULL, MISSING.

Reply:
@snej not hard baking NOT into the operator, I can simply make a new function, as Jens suggested, nullormissing, and JSON would be ["nullormissing()", arg]

Reply:
DM'ed with @pasin, and I will remove IS NOT NULLORMISSING
I will also change ["NULLORMISSING", arg] to ["NULLORMISSING()", arg]

Reply:
It would **not** be "NOT x IS NULLORMISSING"  It would still be "x IS NOT NULLORMISSING" in N1QL even after this change.  If not, then `IS NOT NULL` would not work . Actually I said the wrong operator before.  In this case it wouldn't be the `NOT` operator that got hit, but the infix `IS NOT`.  I think we even made the `IS` optional but have a look [here](https://github.com/couchbase/couchbase-lite-core/blob/master/LiteCore/tests/N1QLParserTest.cc#L171) for what I am imagining.  Just replace `null` with `nullormissing`

Reply:
I'm pretty confusing about what will and will not work now. From the API standpoint, "NOT IS NULLORMISSING" will be pretty confusing to user; if that will be the case, I feel like making it like function will be less confusing.

Reply:
All work. Jim's argument is that we don't have IS NOT NULL, IS NOT MISSING, why do we make it special with IS NOT NULLORMISSING, right, @borrrden

Reply:
Exactly.  It will work the way it is laid out in the spec due to the fact that we already have an `IS NOT` operator present.  The parser doesn't care what comes after it, be it null, missing, nullormissing, or something else entirely